### ISSUE TITLE
doxygen stops parsing a file if one uses latex formulas and `//!<` or `///<` for inline parameter documentation

### ISSUE THREAD

**Issue by lorddavidiii** [doxygen/doxygen#9733]
**Describe the bug**
doxygen stops parsing a file without a warning if one uses inline parameter documentation for a meber function with `//!<` or `///<`  and use latex formulas which is not followed by a whitespace or newline, eg `\f$\sqrt{(x)}\f$-character`. So this function and everything below is missing in the output.

**Expected behavior**
the same behaviour as one gets if one uses `/**< ... */` and `/*!< ... */`. There it is no problem to use latex followed by another letter.

**To Reproduce**
I attached an [minmal example](https://github.com/doxygen/doxygen/files/10269692/minimal_example.tar.gz)

**Version**
output of `doxygen --version`: 1.9.5

output of `lsb_release -a`:
```
LSB Version:	n/a
Distributor ID:	Arch
Description:	Arch Linux
Release:	rolling
Codename:	n/a
```

I use a binary that comes with the distribution

**Comment by lorddavidiii** [comment]
>  the example code is not correct as in the argument lists there is a , that should not be there (does not influence the outcome).

ah right, good catch, I have used `func.h` from the examples, and removed unnecessary things, but miss the `,`.

The real world example is [this line](https://git.ligo.org/lscsoft/lalsuite/-/blob/8a58139b1e0c72f15bcf3548493a36afb8bf9ce8/lalpulsar/lib/ComputeFstat.c#L361) (ie `///< [in] Requested spacing of \f$\mathcal{F}\f$-statistic frequency ...`) in `lalsuite`. So I think this must have worked in some previous doxygen versions. The official `lalsuite` documentation is build with doxygen 1.9.1, which is also broken, see [here](https://lscsoft.docs.ligo.org/lalsuite/lalpulsar/group___compute_fstat__h.html#gacba2c5f40c1f0fb0e6fa6bf4bc9498d6).

**Comment by lorddavidiii** [comment]
So, I also investigate this a bit more, so it seems, that there are more characters without problems, ie also `,.()'` are making no problems.

**Comment by albert-github** [comment]
Correct the characters `A...Z`, `a...z` and `-` give problems.

I've just pushed a proposed patch, pull request #9733

**Comment by lorddavidiii** [comment]
Great, thanks for the quick fix. I just applied the patch on the 1.9.5 version and rebuild the archlinux package with this and can confirm, that this fixes my issue. Thank you!

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.6.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
